### PR TITLE
zero length file upload

### PR DESCRIPTION
**Pull request by godrei**
Updated `TestXCresult3Converters` as it compared structs with a slice property using `reflect.DeepEqual` it was failing from time to time

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -158,6 +160,25 @@ workflows:
...
+    - script:
+        inputs:
+        - content: touch 0_length_file.txt
+    - path::./:
+        title: Zero length file upload test
+        run_if: true
+        inputs:
+        - build_url: $BITRISE_BUILD_URL
+        - build_api_token: $BITRISE_BUILD_API_TOKEN
+        - is_compress: "true"
```
Comment (lpusok) [inline bitrise.yml]: Does is_compress affect file_length in any way?

File: uploaders/common.go [hunk 2]
```diff
@@ -134,11 +147,6 @@ func uploadArtifact(uploadURL, artifactPth, contentType string) error {
...
 		}
 
-		// Set Content Length manually (https://stackoverflow.com/a/39764726), as it is part of signature in signed URL
-		fileInfo, err := file.Stat()
-		if err != nil {
-			return fmt.Errorf("failed to get file info for %s, error: %s", artifactPth, err)
-		}
 		request.ContentLength = fileInfo.Size()
 
 		// The go http library will switch the TransferEncoding to 'chunked' if
```
Comment (lpusok) [inline uploaders/common.go]: Is this part still needed?

File: bitrise.yml (hunk 1 above)
Comment (godrei) [reply bitrise.yml]: pr updated: removed options not affecting this test

File: uploaders/common.go (hunk 2 above)
Comment (godrei) [reply uploaders/common.go]: good catch, I am testing it

File: test/converters/test_utils.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	return reflect.DeepEqual(a, b)
+}
+
+func equivalentTestCases(a, b []junit.TestCase) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	bCopy := append([]junit.TestCase{}, b...)
+	for _, caseA := range a {
```
Comment (lpusok) [inline test/converters/test_utils.go]: Consider testcases that are in B but they are not in A.

**Pull request merged**